### ISSUE TITLE
Registry errors rendering HTML responses not auth server JSON message

### ISSUE THREAD

**Issue by ewindisch** [docker/docker#14367]
Version: Docker master (pre-1.8)

*Description:*

Registry HTML text is printed to user rather than human-readable messages.

*To reproduce:*

Logged in as a non-existent user, I attempt a push:

```
dockerbuilder1:~# docker push adjdjkfjkdjfjk/bob
The push refers to a repository [adjdjkfjkdjfjk/bob] (len: 1)
Sending image list
Error: Status 500 trying to push repository adjdjkfjkdjfjk/bob: "<!doctype html>\n<html class=\"no-js\" lang=\"\">\n    <head>\n        <meta charset=\"utf-8\">\n        <meta http-equiv=\"X-UA-Compatible\" content=\"IE=edge\"><script type=\"text/javascript\">window.NREUM||(NREUM={}),__nr_require=function(e,n,t){
```

A similar error happens on login if the user already exists.

*Expected result:*

Above is the response given by 'GET https://index.docker.io/v1/repositories/library/ubuntu/images', but what should be printed is the response to 'https://auth.docker.io/':

```
{"details":"incorrect username or password"}
```

**Comment by henrysher** [comment]
i met the similiar error and i suggest to handling more error messages in client side.
> FATA[0015] Error response from daemon: </html> (Code: 500; Headers: map[X-Frame-Options:[SAMEORIGIN] Strict-Transport-Security:[max-age=31536000] Server:[nginx/1.6.2] Date:[Fri, 03 Jul 2015 00:44:20 GMT] Content-Type:[text/html; charset=utf-8] Vary:[Cookie]])

**Comment by phemmer** [comment]
There is a PR open to fix this: #14332

**Comment by runcom** [comment]
closing since #14332 has been merged
